src/createServer.js

```diff
@@ -1,3 +1,60 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+
+const http = require('http');
+const { convertToCase } = require('./convertToCase');
+
+const { ServerError } = require('./types/ServerError');
+
+function createServer() {
+  const server = http.createServer((request, response) => {
+    response.setHeader('Content-Type', 'application/json');
+
+    const url = new URL(request.url, 'http://localhost:3000');
+
+    const convertCases = ['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'];
+
+    const { pathname, searchParams } = url;
+
+    const errors = [];
+
+    const text = pathname.slice(1);
+
+    const params = Object.fromEntries(searchParams.entries());
+
+    if (text === '') {
+      errors.push({
+        message: ServerError.EmptyText,
+      });
+    }
+
+    if (Object.keys(params).length === 0) {
+      errors.push({
+        message: ServerError.NoToCase,
+      });
```

Comment:
```suggestion
    if (!params.toCase) {
      errors.push({
        message: ServerError.NoToCase,
      });
```
Your condition is incorrect - it only checks if you have no query params. But if it present it can be anything, not necessary `toCase`.